### DISCUSSION TITLE
diameter estimation, goodness of fit

### DISCUSSION THREAD

**Discussion by akraskovSMDB** [neuropoly/axondeepseg#746] [Help needed]
Hi,

I am wondering whether ADS morphometric calculation can export the goodness of ellipse/circle fit. Below are a couple of examples of concave-shaped axons with estimated diameters (minor axis of the fitted ellipse.
Visually it looks like the diameter is overestimated. I think it looks like this because, given the area and concave perimeter, the minor axes of the best-fitted ellipse will be longer than the smallest distance between myelin walls. 
![image](https://github.com/axondeepseg/axondeepseg/assets/25688017/c469d3be-06a7-450c-89d9-1f2a7b0455f2)
The goodness of fit measure will help me to automatically spot "concave" axons and maybe treat them slightly differently from nice convex/roundish axons.

Thank you

**Comment by mathieuboudreau** [comment]
Hi @akraskovSMDB !

When in "circle" axon mode, we don't fit the axon perimeter to a circle,  and an estimated diameter of a perfect circle that would have the equivalent area as the axon (i.e. we take the area, that is the sum of all segmented axon pixels, and then calculate the equivalent circle).

"Ellipse" axon mode would be more useful in the case of axon fibres that were cut at an angle I believe, which isn't your case.

For you, I think the "concave" axons would result in a low "solidity" value (a measure of "compactness" of the axon) relative to more circular ones. They would also tend to have eccentricity values that are closer to 1 (i.e. more elliptical). Both of these values are reported in the morphometrics file we generate. Please take a look and see if you see a trend for these two values for your concave axons vs the more circular ones.

**Reply by mathieuboudreau** [reply]
@akraskovSMDB did you look into this ?

**Reply by akraskovSMDB** [reply]
Yep, sorry for not responding earlier. Solidity seems to be a reasonable measure for the "non-circularity" of axons.

Just to follow up on the physiological interpretation of the concave axons. It seems that axons are assumed to be perfect cylinders, meaning that their cross-section can be either a perfect circle (perfectly perpendicular cross-section ) or an ellipse with a minor axis perfectly corresponding to the cylinder diameter. 
Most published round axons are small and look round probably because they are taken from rat or mouse tissue.  I almost never see perfectly round axons in my images (humans). Are you aware of any published research on whether non-round/concave axons are just artifact of the tissue preparation procedure (fixation, staining etc ) or axons can actually be non-round concave?

Thank you

**Reply by mathieuboudreau** [reply]
Thanks @akraskovSMDB !

>Are you aware of any published research on whether non-round/concave axons are just artifact of the tissue preparation procedure (fixation, staining etc ) or axons can actually be non-round concave?

Are you aware of this paper? https://www.nature.com/articles/s41598-021-82575-9. See Figure 2 a and b.

**Reply by akraskovSMDB** [reply]
@mathieuboudreau thank you, very useful paper, although it doesn't really answer my question because concave axonal shapes cannot be produced by the oblique cut of the cylinder, something else must be happening with these axons. 
But the reference is still useful, thank you

What method does AxonDeepSeg actually use for the estimation of the minor ellipse axis? Do you use skimage.measure.EllipseModel? It probably gives very similar if not identical to SAE results. Don't you think? 

I guess the solidity measure reported by AxonDeepSeeg morphometrics is similar to Roundness in Bertmeyer et al, but is not exactly the same, is this correct?
